girder_worker/plugins/docker/executor.py

```diff
@@ -118,6 +118,26 @@ def _docker_gc(tempdir):
     return subprocess.Popen(args=(script,), env=env)
 
 
+def validate_task_outputs(task_outputs):
+    """
+    This is called prior to fetching inputs to make sure the output specs are
+    valid. Outputs in docker mode can result in side effects, so it's best to
+    make sure the specs are valid prior to fetching.
+    """
+    for name, spec in task_outputs.iteritems():
+        if spec.get('target') == 'filepath':
+            if 'path' not in spec:
```

Comment:
> Are you asking them to specify the path in the task spec, so people, if needed, can create sub-directories under /data to avoid file name conflicts?

Yes, in fact I'd like them to be able to point to a directory and have the whole thing uploaded to girder recursively if they want.

> Maybe just ask for the relative path and you can internally prepend/join it with /data

I will make it so it works with either option; if an absolute path is specified, it must start with "/data/", but if a relative path is specified it will be assumed as relative to /data.